src/dataflow/src/sink/kafka.rs

```diff
@@ -242,7 +248,7 @@ struct KafkaSinkState {
     topic: String,
     topic_prefix: String,
     shutdown_flag: Arc<AtomicBool>,
-    metrics: SinkMetrics,
+    metrics: Arc<SinkMetrics>,
```

Comment:
This is a neat way to retain d-o-d while keeping them in a struct that has to get cloned a bunch (the only way, really 😄).